File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/offroute/OffRoute.java
package com.mapbox.services.android.navigation.v5.offroute;

import android.location.Location;

import com.mapbox.services.android.navigation.v5.navigation.MapboxNavigationOptions;
import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;

public abstract class OffRoute {

  public abstract boolean isUserOffRoute(Location location, RouteProgress routeProgress,
                                         MapboxNavigationOptions options);
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/offroute/OffRoute.kt
package com.mapbox.services.android.navigation.v5.offroute

import android.location.Location
import com.mapbox.services.android.navigation.v5.navigation.MapboxNavigationOptions
import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress

abstract class OffRoute {

    abstract fun isUserOffRoute(
        location: Location,
        routeProgress: RouteProgress,
        options: MapboxNavigationOptions
    ): Boolean
}
